### PR TITLE
Improve handling of Jaeger connection failures in tests

### PR DESCRIPTION
**Pull request by Yas-Franco**
For issue [#24196](https://github.com/OpenLiberty/open-liberty/issues/24196)

### CODE REVIEW

**Comment by Yas-Franco** [comment]
#build
#libby

**Review by benjamin-confino — APPROVED** [review]

File: dev/io.openliberty.microprofile.telemetry.1.0.internal.jaeger_fat/fat/src/io/openliberty/microprofile/telemetry/internal/utils/jaeger/JaegerQueryClient.java [hunk 1]
```diff
@@ -158,35 +165,38 @@ public List<String> getServices() {
...
-                Thread.sleep(100);
+            } catch (InterruptedException e) {
+                throw new AssertionError("Interrupted while waiting for spans", e);
             }
-        } catch (InterruptedException e) {
-            throw new AssertionError("Iterrupted while waiting for spans", e);
         }
+        assertThat("Spans not found within timeout after 3 retries", result, waitCondition);
+        return result;
     }
```
Comment (Azquelt) [inline dev/io.openliberty.microprofile.telemetry.1.0.internal.jaeger_fat/fat/src/io/openliberty/microprofile/telemetry/internal/utils/jaeger/JaegerQueryClient.java:200]: Why was this changed? I'm not sure why doing a fixed number of retries is better than setting a time limit.
